From c4954f2b9126edccbffa76d860aebf858c80bc4d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 21 May 2016 21:52:36 -0700
Subject: [PATCH] clang: driver: Use /lib for ldso on OE

OE does not follow the default base_libdir
that clang has, therefore adjust it for OE
it wont be able to support multilib since
in multilib case OE switches the base libdir
for 64bit to /lib64 instead of /lib

Make sure that the change only happens for cross
compilation on x86_64 architecture

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 clang/lib/Driver/ToolChains/Linux.cpp | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/clang/lib/Driver/ToolChains/Linux.cpp b/clang/lib/Driver/ToolChains/Linux.cpp
index ed907549929..8cee2490cc7 100644
--- a/clang/lib/Driver/ToolChains/Linux.cpp
+++ b/clang/lib/Driver/ToolChains/Linux.cpp
@@ -593,12 +593,12 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     Loader = "ld.so.1";
     break;
   case llvm::Triple::ppc64:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv2")) ? "ld64.so.2" : "ld64.so.1";
     break;
   case llvm::Triple::ppc64le:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader =
         (tools::ppc::hasPPCAbiArg(Args, "elfv1")) ? "ld64.so.1" : "ld64.so.2";
     break;
@@ -620,7 +620,7 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     Loader = "ld-linux.so.2";
     break;
   case llvm::Triple::sparcv9:
-    LibDir = "lib64";
+    LibDir = "lib";
     Loader = "ld-linux.so.2";
     break;
   case llvm::Triple::systemz:
@@ -633,8 +633,10 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
     break;
   case llvm::Triple::x86_64: {
     bool X32 = Triple.getEnvironment() == llvm::Triple::GNUX32;
-
-    LibDir = X32 ? "libx32" : "lib64";
+    if (Triple.getEnvironment() == llvm::Triple::GNU)
+      LibDir = X32 ? "libx32" : "lib64";
+    else
+      LibDir = "lib";
     Loader = X32 ? "ld-linux-x32.so.2" : "ld-linux-x86-64.so.2";
     break;
   }
